### PR TITLE
RTK and dual GPS usage

### PR DESCRIPTION
**Pull request by priseborough**
Adds information on how to configure the EKF to use RTK GPS for yaw and use of dual GPS.

### CODE REVIEW

File: en/gps_compass/rtk_gps.md [hunk 1]
```diff
@@ -93,6 +93,17 @@ To ensure MAVLink2 is used:
 You may also need to tune some parameters as the default parameters are tuned assuming a GPS accuracy in the order of meters, not centimeters. 
 For example, you can decrease [EKF2_GPS_V_NOISE](../advanced_config/parameter_reference.md#EKF2_GPS_V_NOISE) and [EKF2_GPS_P_NOISE](../advanced_config/parameter_reference.md#EKF2_GPS_P_NOISE) to 0.2.
 
+
+### Use RTK GPS for Yaw
```
Comment (hamishwillee) [inline en/gps_compass/rtk_gps.md]: FYI I moved these up under "Optional Configuraton" - I don't think they needed their own top level heading.

**Comment by hamishwillee** [comment]
@priseborough Looks great. I subedited for wiki style and moved a few bits around. You happy for me to merge, or do you want @thomasgubler to review first?

**Comment by priseborough** [comment]
happy for you to merge

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 2]
```diff
@@ -70,13 +70,43 @@ When tilt and yaw alignment is complete, the EKF can then transition to other mo
...
+
+The following items should be checked during setup:
+
+* Verify that data for the second receiver is present. 
+  This will be logged as `vehicle_gps_position_1` and can also be checked when connected via *nsh console* using the `listener` command.
+* Check the `s_variance_m_s`, `eph` and `epv` data from each receiver and decide which accuracy metrics can be used. 
+  If both receivers output sensible `s_variance_m_s` and `eph` data, and GPS vertical position is not being used directly for navigation, then setting [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) to 3 is recommmeded. 
+  Where only `eph` data is available and both receivers do not output `s_variance_m_s` data, set [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) to 2. 
+  Bit position 2 would only be set if the GPS had been selected as a primary height source with the [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) parameter and both receivers output sensible `epv` data.
+* The output from the blended receiver data is logged as `vehicle_gps_position_` and should be checked.
```
Comment (bkueng) [inline en/advanced_config/tuning_the_ecl_ekf.md]: Should this be `vehicle_gps_position_2`?

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 3]
```diff
@@ -70,13 +70,43 @@ When tilt and yaw alignment is complete, the EKF can then transition to other mo
...
+This is controlled by the [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter. 
+
+The [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter is set by default to disable blending and always use the first receiver, so it will have to be set to select which receiver accuracy metrics are used to decide how much each receiver output contributes to the blended solution. 
+Where different receiver models are used, it is important that the [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter is set to a value that uses accuracy metrics that are supported by both receivers. 
+For example do not set bit position 0 to `true` unless the drivers for both receivers publish values in the `s_variance_m_s` field of the `vehicle_gps_position` message that are comparable. 
+This can be difficult with receivers from different manufacturers due to the different way that accuracy is defined, e.g. CEP vs 1-sigma, etc.
+
+The following items should be checked during setup:
+
+* Verify that data for the second receiver is present.
```
Comment (bkueng) [inline en/advanced_config/tuning_the_ecl_ekf.md]: We can reference `GPS_2_CONFIG`

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 4]
```diff
@@ -70,13 +70,43 @@ When tilt and yaw alignment is complete, the EKF can then transition to other mo
...
+
+The [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter is set by default to disable blending and always use the first receiver, so it will have to be set to select which receiver accuracy metrics are used to decide how much each receiver output contributes to the blended solution. 
+Where different receiver models are used, it is important that the [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) parameter is set to a value that uses accuracy metrics that are supported by both receivers. 
+For example do not set bit position 0 to `true` unless the drivers for both receivers publish values in the `s_variance_m_s` field of the `vehicle_gps_position` message that are comparable. 
+This can be difficult with receivers from different manufacturers due to the different way that accuracy is defined, e.g. CEP vs 1-sigma, etc.
+
+The following items should be checked during setup:
+
+* Verify that data for the second receiver is present. 
+  This will be logged as `vehicle_gps_position_1` and can also be checked when connected via *nsh console* using the `listener` command.
```
Comment (bkueng) [inline en/advanced_config/tuning_the_ecl_ekf.md]: The command would be: `listener vehicle_gps_position -i 1`

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 5]
```diff
@@ -70,13 +70,43 @@ When tilt and yaw alignment is complete, the EKF can then transition to other mo
...
+* Check the `s_variance_m_s`, `eph` and `epv` data from each receiver and decide which accuracy metrics can be used. 
+  If both receivers output sensible `s_variance_m_s` and `eph` data, and GPS vertical position is not being used directly for navigation, then setting [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) to 3 is recommmeded. 
+  Where only `eph` data is available and both receivers do not output `s_variance_m_s` data, set [EKF2_GPS_MASK](../advanced_config/parameter_reference.md#EKF2_GPS_MASK) to 2. 
+  Bit position 2 would only be set if the GPS had been selected as a primary height source with the [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) parameter and both receivers output sensible `epv` data.
+* The output from the blended receiver data is logged as `vehicle_gps_position_` and should be checked.
+* Where receivers output at different rates, the blended output will be at the rate of slower receiver. 
+  Where possible receivers should be configured to output at the same rate.
```
Comment (bkueng) [inline en/advanced_config/tuning_the_ecl_ekf.md]: We should add that you can check the output with `listener ekf_gps_position`

Comment (priseborough) [reply en/advanced_config/tuning_the_ecl_ekf.md]: See 
https://github.com/PX4/px4_user_guide/pull/444/commits/529ca0f2356f2bb9beac9a247560d3a683cc7307

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 2 above)
Comment (priseborough) [reply en/advanced_config/tuning_the_ecl_ekf.md]: The output is now logged as ekf_gps_position. I will update the documentation accordingly. See 
https://github.com/PX4/px4_user_guide/pull/444/commits/529ca0f2356f2bb9beac9a247560d3a683cc7307

**Pull request merged**